modules/cli/internal/sequencer/sequencer.go

```diff
@@ -783,33 +694,24 @@ func ChangeSudoAddress(opts ChangeSudoAddressOpts) (*ChangeSudoAddressResponse,
 	log.Debugf("Change Sudo Address Opts: %v", opts)
 
 	// client
-	opts.SequencerURL = addPortToURL(opts.SequencerURL)
 	log.Debug("Creating CometBFT client with url: ", opts.SequencerURL)
 	c, err := client.NewClient(opts.SequencerURL)
 	if err != nil {
 		log.WithError(err).Error("Error creating sequencer client")
 		return &ChangeSudoAddressResponse{}, err
 	}
 
-	// create signer
-	from, err := privateKeyFromText(opts.FromKey)
-	if err != nil {
-		log.WithError(err).Error("Error decoding private key")
-		return &ChangeSudoAddressResponse{}, err
-	}
-	signer := client.NewSigner(from)
-
 	// Get current address nonce
+	signer := client.NewSigner(opts.FromKey)
```

Comment:
I think we should leave this for now. Both the `signer` and the `addr` that get made in these sections get used in other places in the functions. So it will be a little more of a refactor that just moving the logic.